integration-tests/contracts/ethereum_keeper_contracts.go

```diff
@@ -781,6 +826,178 @@ func (o *KeeperConsumerPerformanceRoundConfirmer) logDetails() {
 	defer o.metricsReporter.ReportMutex.Unlock()
 }
 
+// KeeperConsumerBenchmarkRoundConfirmer is a header subscription that awaits for a round of upkeeps
+type KeeperConsumerBenchmarkRoundConfirmer struct {
+	instance KeeperConsumerBenchmark
+	upkeepID *big.Int
+	doneChan chan bool
+	context  context.Context
+	cancel   context.CancelFunc
+
+	blockRange      int64                                      // How many blocks to watch upkeeps for
+	upkeepSLA       int64                                      // SLA after which an upkeep is counted as 'missed'
+	metricsReporter *testreporters.KeeperBenchmarkTestReporter // Testreporter to track results
+
+	// State variables, changes as we get blocks
+	blocksSinceSubscription int64   // How many blocks have passed since subscribing
+	blocksSinceEligible     int64   // How many blocks have come in since upkeep has been eligible for check
+	countEligible           int64   // Number of times the upkeep became eligible
+	countMissed             int64   // Number of times we missed SLA for performing upkeep
+	upkeepCount             int64   // The count of upkeeps done so far
+	allCheckDelays          []int64 // Tracks the amount of blocks missed before an upkeep since it became eligible
+	complete                bool
+}
+
+// NewKeeperConsumerBenchmarkRoundConfirmer provides a new instance of a KeeperConsumerBenchmarkRoundConfirmer
+// Used to track and log benchmark test results for keepers
+func NewKeeperConsumerBenchmarkRoundConfirmer(
+	contract KeeperConsumerBenchmark,
+	upkeepID *big.Int,
+	blockRange int64,
+	upkeepSLA int64,
+	metricsReporter *testreporters.KeeperBenchmarkTestReporter,
+) *KeeperConsumerBenchmarkRoundConfirmer {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	return &KeeperConsumerBenchmarkRoundConfirmer{
+		instance:                contract,
+		upkeepID:                upkeepID,
+		doneChan:                make(chan bool),
+		context:                 ctx,
+		cancel:                  cancelFunc,
+		blockRange:              blockRange,
+		upkeepSLA:               upkeepSLA,
+		blocksSinceSubscription: 0,
+		blocksSinceEligible:     0,
+		upkeepCount:             0,
+		allCheckDelays:          []int64{},
+		metricsReporter:         metricsReporter,
+		complete:                false,
+	}
+}
+
+// ReceiveBlock will query the latest Keeper round and check to see whether the round has confirmed
+func (o *KeeperConsumerBenchmarkRoundConfirmer) ReceiveBlock(receivedBlock blockchain.NodeBlock) error {
+	// Increment block counters
+	o.blocksSinceSubscription++
```

Comment:
I saw this PR (https://github.com/smartcontractkit/chainlink/pull/7144/files) which accounts for reorgs here. Can also add it here as a followup

Reply:
Thanks for pointing this out. This was meant to be copied here since it affects test execution on Goerli.